set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java

```diff
@@ -279,6 +287,32 @@
 				// validate and process the Set description for format and mandatory fields
 				// if false respond with HTTP 400
 				getUserSetService().validateWebUserSet(newUserSet);
+				
+				// if the Set corresponds to an open set, update the metadata associated to the Set, 
+				// and if the standard profile is requested, obtain the URIs for the items following 
+				// the logic defined for retrieving a Set.
+				if (getUserSetService().isOpenSet(newUserSet)) {
+					newUserSet = updateItemsWithIsDefinedBy(newUserSet,
+							null, null, WebUserSetFields.DEFAULT_PAGE, WebUserSetFields.MAX_ITEMS_PER_PAGE);
+				} else {
+					// If the provided userset contains a list of items and the profile is set to minimal, 
+					// respond with HTTP 412, also when standard profile is used and no items are provided, 
+					// respond with 412;
+					LdProfiles headerProfile = getHeaderProfile(request);
+					if ((newUserSet.getItems() != null && newUserSet.getItems().size() > 0 && (profile == LdProfiles.MINIMAL || headerProfile == LdProfiles.MINIMAL)) || 
+					    ((newUserSet.getItems() == null || newUserSet.getItems().size() == 0) && profile == LdProfiles.STANDARD)) { 
+						throw new ApplicationAuthenticationException(
+							I18nConstants.USERSET_CONTAINS_NO_ITEMS, I18nConstants.USERSET_CONTAINS_NO_ITEMS,
+							new String[] {}, HttpStatus.PRECONDITION_FAILED, null);	
+					}
+
+					// if the Set corresponds to a closed set, replace member items with the new items 
+					// that are present in the Set description only when a profile is indicated and 
+					// is different from "ldp:PreferMinimalContainer" is referred in the "Prefer" header.
+					if (getHeaderProfile(request) != LdProfiles.MINIMAL)
+						newUserSet = getUserSetService().updateUserSetsWithCloseSetItems(existingUserSet, newUserSet.getItems());
+				}
+				
```

Comment:
the generation of the eTag should happen after updating the user set into database, see also the updates to specification

Reply:
improved logic according to suggestion and specification